lib/MumbleConnection.js

```diff
@@ -366,6 +386,36 @@ MumbleConnection.prototype._processMessage = function( type, msg ) {
 };
 
 /**
+ * Handle codecVersion message
+ *
+ * @param msg Codec message
+ **/
+MumbleConnection.prototype._onCodecVersion = function( codecVersion ) {
+    if(this.opusEnabled && ! codecVersion.opus) { // Server switched from opus to celt
+        for(var key in this.users) {
+            var user = this.users[key];
+            if( user ){
+                user.decoder = new CeltEncoder( this.SAMPLING_RATE, this.FRAME_SIZE );
+            }
+        }
+    }
+    if(!this.opusEnabled && codecVersion.opus && this.options.opus) { // Server switched from celt to opus
+        for(var key in this.users) {
+            var user = this.users[key];
+            if( user ){
+                user.decoder = new OpusEncoder( this.SAMPLING_RATE );
```

Comment:
Think we should have both opus and celt encoder for each user - possibly on demand.